### DISCUSSION TITLE
Clarification on the Advantages of Using Variants vs. Normal Inherited Objects

### DISCUSSION THREAD

**Discussion by AlephLau** [pimcore/pimcore#16856] [Q&A / Support] [answered]
Hello Pimcorenauts,

My main question revolves around the specific advantages of using object variants over normal objects (children) with inheritance. The documentation highlights that a key difference between object variants and normal inherited objects is the ability to hide variants from the object tree. However, since inheritance can be achieved without necessarily using the variant type, I'm wondering if there are other, perhaps less obvious, benefits to using variants.

Specifically, I'm interested in understanding:

Are there any performance optimizations, at the database level or otherwise, when using variants instead of simply relying on inheritance for similar objects?
Does the use of variants impact the speed or efficiency of data manipulation or retrieval in any significant way?
Are there any other practical benefits or use cases where variants provide a clear advantage over normal inherited objects that might not be immediately apparent?
Any insights, experiences, or examples from the community or the Pimcore team would be greatly appreciated. I'm looking to ensure that our data model is as efficient and scalable as possible, and understanding these nuances could greatly influence our approach.

Thank you in advance for your time and assistance.

**Comment by AlephLau** [comment]
@brusch  @fashxp Could you clarify this for me?

**Reply by fashxp** [reply, marked as answer]
technically data objects and variants are the same ... it is just a flag, that decides the type.

**Reply by AlephLau** [reply]
Thanks @fashxp 🥇
